File: 【前缀和】2023B-分割数组的最大差值.cpp
/*
 * @hw id=2023B lang=cpp
 *
 * 分割数组的最大差值
 */

// @hw code=start
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

int main() {
    int n;
    cin >> n;
    int nums[n];
    int lsum = 0, rsum = 0, ans = 0;
    for(int i=0;i<n;i++) {
        cin >> nums[i];
        rsum += nums[i];
    }
    for(int i=0;i<n;i++) {
        lsum += nums[i];
        rsum -= nums[i];
        ans = max(ans, abs(lsum - rsum));
    }
    cout << ans << endl;
    return 0;
}
// @hw code=end

File: HW/【前缀和】2023B-分割数组的最大差值.java
/*
 * @hw id=2023B lang=java
 *
 * 分割数组的最大差值
 */

// @hw code=start
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] nums = new int[n];
        int rsum = 0, lsum = 0, ans = 0;
        for(int i=0;i<n;i++) {
            nums[i] = in.nextInt();
            rsum += nums[i];
        }
        for(int i=0;i<n;i++) {
            lsum += nums[i];
            rsum -= nums[i];
            ans = Math.max(ans, Math.abs(lsum - rsum));
        }
        System.out.println(ans);
    }
}
// @hw code=end